packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/auth/EmailPasswordAuth.kt

```diff
@@ -87,18 +87,17 @@ public interface EmailPasswordAuth {
      */
     public suspend fun sendResetPasswordEmail(email: String)
 
-    // TODO https://github.com/realm/realm-kotlin/issues/744
-    // /**
-    //  * Call the reset password function configured to the
-    //  * [Credentials.Provider.EMAIL_PASSWORD] provider.
-    //  *
-    //  * @param email the email of the user.
-    //  * @param newPassword the new password of the user.
-    //  * @param args any additional arguments provided to the reset function. All arguments must
-    //  * be able to be converted to JSON compatible values using `toString()`.
-    //  * @throws AppException if the server failed to confirm the user.
-    //  */
-    // public suspend fun callResetPasswordFunction(email: String, newPassword: String, vararg args: Any?)
+    /**
+     * Call the reset password function configured to the
+     * [Credentials.Provider.EMAIL_PASSWORD] provider.
+     *
+     * @param email the email of the user.
+     * @param newPassword the new password of the user.
+     * @param args any additional arguments provided to the reset function. All arguments must
+     * be able to be converted to JSON compatible values using `toString()`.
+     * @throws AppException if the server failed to confirm the user.
```

Comment:
Is this true? Similar to the discussion here https://github.com/realm/realm-kotlin/pull/1131/files#r1034502924 we should probably return an `AuthException` if there is an error in the function?

Reply:
it is actually `ServiceException`, which I tried to change but now that I think about it makes sense perhaps since this is (in theory) an issue with the reset password service and not the authentication (at least that's why I assume this is the error? I know `AuthException` is a subset of that but it does fail if I try to assert specifically that)